common/util/autoware_point_types/include/autoware_point_types/types.hpp

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 Tier IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef AUTOWARE_POINT_TYPES__AUTOWARE_POINT_TYPES_HPP_
+#define AUTOWARE_POINT_TYPES__AUTOWARE_POINT_TYPES_HPP_
+
+#include <point_cloud_msg_wrapper/point_cloud_msg_wrapper.hpp>
+
+#include <cmath>
+#include <limits>
+#include <tuple>
+
+namespace autoware_point_types
+{
+template <class T>
+bool approximate_equal(const T a, const T b, const T eps = std::numeric_limits<T>::epsilon())
+{
+  const auto a_abs = std::fabs(a);
+  const auto b_abs = std::fabs(b);
+  return std::fabs(a - b) <= (a_abs < b_abs ? b_abs : a_abs) * eps;
```

Comment:
- Why do you not use `std::max`?
- Why do you multiply `max of a and b` and `eps`, not use just `eps`?
  - I feel unnatural to this behavior. https://wandbox.org/permlink/Trxt5irtm3XFHbkz

Reply:
> Why do you multiply max of a and b and eps, not use just eps?

I had have same question, but according to this article, considering the digit loss error, it seems that this way is good.
https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/

Reply:
Then I guess you should input `percent` instead of `epsilon`. :thinking:

Reply:
The percent needs to be determined appropriately depending on the compare values, but in many cases FLT_EPSILON seems to work well, so I did.

> When doing a relative comparison of floats it works pretty well to set maxRelDiff to FLT_EPSILON

Reply:
> Why do you not use std::max?

I fixed it in https://github.com/tier4/AutowareArchitectureProposal.iv/pull/784/commits/c26a5bba7317685161fef0c0ddc3c8186c6eeb0d

Reply:
What do you think about https://wandbox.org/permlink/Trxt5irtm3XFHbkz?
I feel you should clarify the specification of this function by adding more tests.

Reply:
I fixed in https://github.com/tier4/AutowareArchitectureProposal.iv/pull/784/commits/2c78c90e547a44c3f984100b53101db0212d4be4.

I thought about a lot but as you said relative comparison is not suitable for this problem. So I changed comaprison method to absolute one and set 10e-6 to epsilon because it's enough to whether point is close.